### PR TITLE
[new release] eio-ssl (0.1.0)

### PR DESCRIPTION
**Pull request by anmonteiro**
OpenSSL binding to EIO

- Project page: <a href="https://github.com/anmonteiro/eio-ssl">https://github.com/anmonteiro/eio-ssl</a>

##### CHANGES:

- Initial public release

### CODE REVIEW

File: packages/eio-ssl/eio-ssl.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+maintainer: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+authors: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+license: "LGPL with OpenSSL linking exception"
+homepage: "https://github.com/anmonteiro/eio-ssl"
+bug-reports: "https://github.com/anmonteiro/eio-ssl/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "5.0"}
+  "ssl" {>= "0.5.13"}
+  "eio"
```
Comment (mseri) [inline packages/eio-ssl/eio-ssl.0.1.0/opam:12]:
```suggestion
  "eio" {> "0.1"}
```

File: packages/eio-ssl/eio-ssl.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+maintainer: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+authors: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+license: "LGPL with OpenSSL linking exception"
+homepage: "https://github.com/anmonteiro/eio-ssl"
+bug-reports: "https://github.com/anmonteiro/eio-ssl/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "5.0"}
+  "ssl" {>= "0.5.13"}
+  "eio" {> "0.1"}
```
Comment (mseri) [inline packages/eio-ssl/eio-ssl.0.1.0/opam:12]:
```suggestion
  "eio" {> "0.3"}
```

Comment (mseri) [reply packages/eio-ssl/eio-ssl.0.1.0/opam]: For
```
#=== ERROR while compiling eio-ssl.0.1.0 ======================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-variants.5.0.0+trunk | pinned(https://github.com/anmonteiro/eio-ssl/releases/download/0.1.0/eio-ssl-0.1.0.tbz)
# path                 ~/.opam/5.0/.opam-switch/build/eio-ssl.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p eio-ssl -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/eio-ssl-8-b4ea37.env
# output-file          ~/.opam/log/eio-ssl-8-b4ea37.out
### output ###
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.eio_ssl.objs/byte -I /home/opam/.opam/5.0/lib/bigarray-compat -I /home/opam/.opam/5.0/lib/cstruct -I /home/opam/.opam/5.0/lib/ctypes -I /home/opam/.opam/5.0/lib/eio -I /home/opam/.opam/5.0/lib/eio/unix -I /home/opam/.opam/5.0/lib/eio/utils -I /home/opam/.opam/5.0/lib/eio_linux -I /home/opam/.opam/5.0/lib/eio_luv -I /home/opam/.opam/5.0/lib/eio_main -I /home/opam/.opam/5.0/lib/fmt -I /home/opam/.opam/5.0/lib/integers -I /home/opam/.opam/5.0/lib/logs -I /home/opam/.opam/5.0/lib/luv -I /home/opam/.opam/5.0/lib/luv/c -I /home/opam/.opam/5.0/lib/luv/c_function_descriptions -I /home/opam/.opam/5.0/lib/luv/c_type_descriptions -I /home/opam/.opam/5.0/lib/luv_unix -I /home/opam/.opam/5.0/lib/lwt -I /home/opam/.opam/5.0/lib/lwt-dllist -I /home/opam/.opam/5.0/lib/mtime -I /home/opam/.opam/5.0/lib/mtime/os -I /home/opam/.opam/5.0/lib/ocaml/threads -I /home/opam/.opam/5.0/lib/ocaml/unix -I /home/opam/.opam/5.0/lib/optint -I /home/opam/.opam/5.0/lib/psq -I /home/opam/.opam/5.0/lib/result -I /home/opam/.opam/5.0/lib/seq -I /home/opam/.opam/5.0/lib/ssl -I /home/opam/.opam/5.0/lib/stdlib-shims -I /home/opam/.opam/5.0/lib/uring -no-alias-deps -o src/.eio_ssl.objs/byte/eio_ssl.cmi -c -intf src/eio_ssl.mli)
# File "src/eio_ssl.mli", line 37, characters 20-41:
# 37 |   val get_fd : t -> Eio.Net.stream_socket
#                          ^^^^^^^^^^^^^^^^^^^^^
# Error: Unbound type constructor Eio.Net.stream_socket
```

**Comment by mseri** [comment]
@kit-ty-kate what is the syntax for custom licenses?

File: packages/eio-ssl/eio-ssl.0.1.0/opam [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+maintainer: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+authors: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+license: "LGPL with OpenSSL linking exception"
+homepage: "https://github.com/anmonteiro/eio-ssl"
+bug-reports: "https://github.com/anmonteiro/eio-ssl/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "5.0"}
+  "ssl" {>= "0.5.13"}
+  "eio" {> "0.3"}
```
Comment (mseri) [inline packages/eio-ssl/eio-ssl.0.1.0/opam:12]:
```suggestion
  "eio" {> "0.4"}
```
Fur to incompatibility with current ocaml beta

File: packages/eio-ssl/eio-ssl.0.1.0/opam [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+authors: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+license: "LGPL with OpenSSL linking exception"
+homepage: "https://github.com/anmonteiro/eio-ssl"
+bug-reports: "https://github.com/anmonteiro/eio-ssl/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "5.0"}
+  "ssl" {>= "0.5.13"}
+  "eio" {> "0.4"}
+  "eio_main"
```
Comment (kit-ty-kate) [inline packages/eio-ssl/eio-ssl.0.1.0/opam:13]: `>` should be avoided in favour of `>=` in most cases.
```suggestion
  "eio" {>= "0.5"}
  "eio_main" {>= "0.5"}
```

**Comment by kit-ty-kate** [comment]
> @kit-ty-kate what is the syntax for custom licenses?

`LicenseRef-your-license-name`

File: packages/eio-ssl/eio-ssl.0.1.0/opam [hunk 5]
```diff
@@ -0,0 +1,38 @@
+opam-version: "2.0"
+synopsis: "OpenSSL binding to EIO"
+maintainer: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+authors: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
+license: "LGPL with OpenSSL linking exception"
```
Comment (mseri) [inline packages/eio-ssl/eio-ssl.0.1.0/opam:5]:
```suggestion
license: "LGPL WITH LicenseRef-OpenSSL-linking-exception"
```

File: packages/eio-ssl/eio-ssl.0.1.0/opam [hunk 6]
```diff
@@ -2,15 +2,15 @@ opam-version: "2.0"
 synopsis: "OpenSSL binding to EIO"
 maintainer: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
 authors: ["Antonio Nuno Monteiro <anmonteiro@gmail.com>"]
-license: "LGPL with OpenSSL linking exception"
+license: "LGPL WITH LicenseRef-OpenSSL-linking-exception"
```
Comment (kit-ty-kate) [inline packages/eio-ssl/eio-ssl.0.1.0/opam:5]:
```suggestion
license: "LicenseRef-LGPL-WITH-OpenSSL-linking-exception"
```

**Comment by mseri** [comment]
The CI is still unhappy about the license, but I think we can merge

**Comment by kit-ty-kate** [comment]
Yes that's a CI issue. I'm on it: https://github.com/ocurrent/opam-repo-ci/pull/186

**Comment by anmonteiro** [comment]
Thank you both! I'll submit these fixes upstream

**Pull request merged**